mysqld_exporter.go

```diff
@@ -601,6 +632,55 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
 			}
 		}
 	}
+
+	if *userStat {
+		informationSchemaUserStatisticsRows, err := db.Query("SELECT * FROM information_schema.USER_STATISTICS")
+		if err != nil {
+			log.Println("Error running user statistics query on database:", err)
+			e.error.Set(1)
+			return
+		}
+		defer informationSchemaUserStatisticsRows.Close()
+
+		var columnNames []string
+		columnNames, err = informationSchemaUserStatisticsRows.Columns()
+		if err != nil {
+			log.Println("Error retrieving column list for information_schema.USER_STATISTICS:", err)
+			e.error.Set(1)
+			return
+		}
+
+		var user string // Holds the username, which should be in column 0
+		var userStatData = make([]float64, len(columnNames))
+		var userStatScanArgs = make([]interface{}, len(columnNames))
+		userStatScanArgs[0] = &user
+		for i := range userStatData[1:] {
+			userStatScanArgs[i+1] = &userStatData[i+1]
+		}
```

Comment:
I would add a comment above this block and explain what's happening  with those slices here and below.